### PR TITLE
PerChannelBookieClient Enhancement: Self-Recovery Support in Corrupte…

### PR DESCRIPTION
**Pull request by shoothzj**
### Motivation
In the context of popular protocols like LengthBasedFrame, which typically determine total message length based on the first 0-4 bytes (for example, if the initial bytes are 00000014, the message length is 1*16+4 = 20), message tampering can cause serious issues. Specifically, any modification to the higher order bytes can lead to miscalculation of message length and leave the communication end waiting indefinitely without any ability to self-recover.

In summary, in scenarios with message tampering, communication timeout can lead to an unrecoverable link.
I think we can rebuild the client when timeout.

### Changes
rebuild the **PerChannelBookieClient** when timeout.

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java [hunk 1]
```diff
@@ -1043,6 +1043,9 @@ public void checkTimeoutOnPendingOperations() {
         if (timedOutOperations > 0) {
             LOG.info("Timed-out {} operations to channel {} for {}",
                      timedOutOperations, channel, bookieId);
+            // In scenarios with message tampering, communication timeout can lead to an unrecoverable state.
+            this.disconnect();
```
Comment (hangc0276) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:1047]: If the timeout is not caused by message tampering, will the reconnect operation cause any side effects?

**Review by hangc0276 — CHANGES_REQUESTED** [review]
Do we need any tests to cover your case?
Need more eyes @eolivelli @dlg99 @merlimat @zymap

**Comment by Shoothzj** [comment]
@hangc0276 Yes, I'm still working on it.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java (hunk 1 above)
Comment (Shoothzj) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:1047]: @hangc0276 I believe there shouldn't be an issue here. However, if everyone feels that its impact is significant or it's too sensitive, we might consider adding a threshold for reconnection.

**Comment by shoothzj** [comment]
Recently, I have no time to work on this, closed for now.
